pkg/orchestrator/errors.go

```diff
@@ -33,16 +35,22 @@ func NewErrNotEnoughNodes(requestedNodes int, availableNodes []NodeRank) ErrNotE
 }
 
 func (e ErrNotEnoughNodes) Error() string {
-	nodeErrors := ""
-	available := 0
-	for _, rank := range e.AvailableNodes {
-		if rank.MeetsRequirement() {
-			available += 1
+	suitable := lo.CountBy(e.AvailableNodes, func(rank NodeRank) bool { return rank.MeetsRequirement() })
+	reasons := lo.GroupBy(e.AvailableNodes, func(rank NodeRank) string { return rank.Reason })
+
+	var message strings.Builder
+	fmt.Fprint(&message, "not enough nodes to run job. ")
+	fmt.Fprintf(&message, "requested: %d, available: %d, suitable: %d.", e.RequestedNodes, len(e.AvailableNodes), suitable)
+	for reason, nodes := range reasons {
+		fmt.Fprint(&message, "\n• ")
+		if len(nodes) > 1 {
+			fmt.Fprintf(&message, "%d of %d nodes", len(nodes), len(e.AvailableNodes))
 		} else {
-			nodeErrors += fmt.Sprintf("\n\tNode %s: %s", idgen.ShortNodeID(rank.NodeInfo.ID()), rank.Reason)
+			fmt.Fprintf(&message, "Node %s", idgen.ShortNodeID(nodes[0].NodeInfo.ID()))
 		}
+		fmt.Fprintf(&message, ": %s", reason)
 	}
-	return fmt.Sprintf("not enough nodes to run job. requested: %d, available: %d. %s", e.RequestedNodes, available, nodeErrors)
+	return message.String()
```

Comment:
generally a fan of this, but we shouldn't be including newlines in our errors.

Reply:
It's currently the only way to get newlines to appear in the CLI, which enables the screenshot above. We _do_ want to use newlines there, and for ranking errors this PR does not include a way to structurally store errors in the job or execution state, which is where this error will end up. Assuming we make this change in a future PR, we can remove the newlines here (and most of this method).